[Fix] 예외처리

david-parkk opened the pull request:
### ✏️ 작업 개요
- #60 

### ⛳ 작업 분류
- [x] 유저
- [x] 캐릭터
- [x] 인벤토리
- [x] 퀘스트
- [x] 랭킹
- [x] 상점

### 🔨 작업 상세 내용
  1.  exception을 정리합니다
  2. 클라에게 보여질 메세지에 대한 범위를 결정하고 있습니다
  3. CustomException 상속해서 만들면됩니다
  4. CusteomException의 다양한 생성자중에 String만 받는 메세지는 G500으로 처리하도록 생성자를 추가하였습니다
  5. 유저 도메인 진행중 몇가지 사용하지 않는 엔드포인트가 있네요. 확인 해봅시다

### 💡 생각해볼 문제
- 예외를 서버 사이드용으로 작성한 부분이 있었는데(resourceNotFound) 현재 ExceptionHandler으로 Exception에 대한 역활이 Client에게 옮겨졌기 때문에 클라입장에서 작성하는게 맞을까요?
- 더 좋은 방법이 있을까요? (클라에게는 안전한 메세지, 서버에게는 서버의 상세 메세지)
- 로깅을 하면 좋겠다는 생각은... 듭니다
- **머지하실 필요없이 동일한 브랜치에서 쭉 작업할 계획입니다~**

david-parkk committed:
fix: application.yml 수정

- redis password 추가
- local 테스트를 위해 추가

david-parkk committed:
feat: CustomException 생성자 추가

- message만으로 생성자 생성가능
- message만 호출시 500으로 판단

david-parkk committed:
feat: G400 ErrorCode(INVALID_INPUT) 추가

david-parkk committed:
fix: UserExistsException 관련 수정

david-parkk committed:
feat: A400 분기

david-parkk committed:
fix: UserNotFoundException 관련 수정

- ResourceNotFoundException -> UserNotFoundException
- exception 처리
- 주석 추가

kamothi commented on src/main/java/ku/user/domain/user/controller/UserController.java:
```diff
@@ -36,12 +36,14 @@ public ResponseEntity<String> success(HttpServletRequest request,HttpServletResp
         return ResponseEntity.status(HttpStatus.OK).body("ok");
     }
 
+    //TODO 사용하는가
```
초기 개발에서 테스트를 위한 쓰레기 코드입니다. 삭제 진행하겠습니다.

kamothi commented on src/main/java/ku/user/domain/user/controller/UserController.java:
```diff
@@ -36,12 +36,14 @@ public ResponseEntity<String> success(HttpServletRequest request,HttpServletResp
     }
 
 
+    //TODO requestBody가 의도된 것인가
```
위와 마찬가지로 현재 사용중이지 않은 테스트용 api였습니다. 삭제 진행하겠습니다.

kamothi reviewed (commented):
> 예외를 서버 사이드용으로 작성한 부분이 있었는데(resourceNotFound) 현재 ExceptionHandler으로 Exception에 대한 역활이 Client에게 옮겨졌기 때문에 클라입장에서 작성하는게 맞을까요?

현재 생각으로는 클라입장이 맞는 것 같습니다.

> 로깅을 하면 좋겠다는 생각은... 듭니다

현재 가지고 있던 생각은 로깅 처리를 진행하고 클라에게는 단편적인 예외 메시지를 보내고 내부 로직에서의 상세 메시지들은 로깅을 통해 모니터링(그라파나)에 세팅을 해둘 생각이였습니다.

david-parkk committed:
feat: QuestNotFoundException 추가

- RuntimeException에서 CustomException으로 변경

david-parkk committed:
fix: 예외 추가

- AlreadyExistCharacterException, DuplicateNicknameException
- CustomException 작성

david-parkk committed:
del: 불필요한 예외 클래스 삭제

david-parkk committed:
fix: NotFoundCharacterException 추가

- CustomException 구현

david-parkk committed:
refactor: 온점 추가

david-parkk committed:
fix: NotFoundInventoryException 추가

- CustomException 구현

david-parkk committed:
fix: NotFoundItemException 추가

david-parkk commented:
RuntimeException 모두 CustoemException으로 구현체 만들었습니다. 모두 ErrorCode로 만들지 않고 최대한 유연하게 대응할 수 있게 만들었는데 그렇게 보일지 모르겠습니다. 
예외코드도 직접 정의하는 방식에서 최대한 HTTP status Code에 근거해서 만들었습니다.
네이밍이 몇개 깨지는게 보이는데 일단 보류했습니다(요새 이런것만 보다보니..)

kamothi reviewed (approved):
수고하셨습니다!
